### ISSUE TITLE
LP1712301: Need S3 stress test for discrete graphic

### ISSUE THREAD

**Issue by beliaev-maksim** [canonical/checkbox#60]
This issue was migrated from https://bugs.launchpad.net/plainbox-provider-checkbox/+bug/1712301

## Summary
| Status 	| Created on 	| Heat 	| Importance 	| Security related 	|
|--------	|------------	|------	|------------	|------------------	|
| Confirmed	| 2017-08-22 09:59:39	| 14	| High	| False  	|

## Description
We do have S3 single test for discrete graphic (Nvidia graphic) but no stress test. It would be better to have S3 stress for Nvidia and AMD dGPUs as it is cert blocker.

How to switch graphic
To switch to Nvidia card:
1. sudo prime-select nvidia
2. reboot system
3. prime-select query to check which card is being used right now

To switch to Intel card:
sudo prime-select intel

For amdgpu-pro stack, it is also switchable (amdgpu is not). We need S3 stress test for amdgpu-pro stack as well. Checkbox already has a way to identify amdgpu and amdgpu-pro.

To switch graphic mode on amdgpu-pro stack
sudo /opt/amdgpu-pro/bin/amdgpu-pro-px --mode performance/powersaving

The idea is to have S3 stress test for discrete graphic in stress test plan and can be completed without interaction.
The critical part is system needs reboot after switching graphic mode (performance-discrete and power saving-UMA).

## Attachments
[/usr/share/plainbox-provider-checkbox/units/graphics/test-plan.pxu](https://api.launchpad.net/devel/bugs/1712301/+attachment/5230185/data)

Tags:
['ce-qa-concern']

**Comment by beliaev-maksim** [comment]
This thread was migrated from launchpad.net
##### https://launchpad.net/~kissiel wrote on 2017-12-03 17:12:18:
Fix landed here: https://code.launchpad.net/~kissiel/plainbox-provider-checkbox/+git/plainbox-provider-checkbox/+merge/333607

##### https://launchpad.net/~ray.chen wrote on 2019-01-18 09:13:00:
This improvement was verified and test on Somerville-graphic test plan after manual replace all 
template job id: graphics/{index}_switch_card_{product_slug}_xenial 
to 
id: graphics/{index}_auto_switch_card_{product_slug} in graphic nested test plan

eg:
"graphics/1_switch_card_.*_xenial " to " graphics/1_auto_switch_card_.*   "




##### https://launchpad.net/~ray.chen wrote on 2019-04-25 08:55:25:
The fix was only fixed dual graphic auto switch, OEM projects need a suspend_30_cycles_with_reboots/logs for dual graphic like 

1. Add new “suspend 30 with reboot” template for dual graphics based on  /stress/jobs.pxu #68~#83 to /suspend/suspend-graphics.pxu
unit: template
template-resource: graphics_card
template-filter: graphics_card.prime_gpu_offload == 'Off' 
id: power-management/{index}_suspend_30_cycles_with_reboots_{product_slug}
plugin: shell
category_id: com.canonical.plainbox::stress
depends:
 power-management/rtc
 suspend/suspend_advanced
 graphics/{index}_auto_switch_card_{product_slug}
requires: ……
flags: noreturn autorestart
user: root
environ: PM_TEST_DRY_RUN
command: …..


2. Add new nested plan for dual graphics into  /stress/test-plan.pxu for example
Id: com.canonical.certification::stress-suspend-30-cycles-with-reboots-automated-dual-graphics 
unit: test plan
_name: Suspend stress tests (with reboots) for dual graphics
_description: Suspend stress tests (with reboots) for dual graphics
include:
power-management/1_suspend_30_cycles_with_reboots_.*
power-management/1_suspend-30-cycle-log-attach-with-reboots_.*
…...
power-management/2_suspend_30_cycles_with_reboots_.*
power-management/2_suspend-30-cycle-log-attach-with-reboots_.*
….


##### https://launchpad.net/~ray.chen wrote on 2019-04-25 08:56:13:
# reopen this issue based on https://docs.google.com/document/d/1Cj-HMAnY-W1uwoYIgTW36aFz3fIbQzVzjKpLktIt9Is/edit#heading=h.4lu4w12akx1h

##### https://launchpad.net/~jocave wrote on 2019-04-25 09:08:41:
Discussion on irc, the effective operations that checkbox should carry out are:

switch to iGPU, reboot, 30 S3, reboot, 30 S3, reboot, 30 S3, attach logs
switch to dGPU, reboot, 30 S3, reboot, 30 S3, reboot, 30 S3, attach logs

**Comment by kissiel** [comment]
It's related to old way of running tests so I think this is no longer relevant.
